benchmarks/cpp/tensorexpr/tensorexpr.cpp

```diff
@@ -0,0 +1,67 @@
+#include <benchmark/benchmark.h>
+#include "torch/torch.h"
+#include "torch/csrc/jit/tensorexpr/ir_simplifier.h"
+#include "torch/csrc/jit/tensorexpr/loopnest.h"
+#include "torch/csrc/jit/tensorexpr/tensor.h"
+
+namespace te = torch::jit::tensorexpr;
+
+class Gemm : public benchmark::Fixture {
+ public:
+  void SetUp(const benchmark::State& state) {
+    M = state.range(0);
+    N = state.range(1);
+    K = state.range(2);
+    A = torch::randn({M, K});
+    B = torch::randn({K, N});
+    C = torch::mm(A, B);
+  }
+
+  void TearDown(benchmark::State& state) {
+    state.counters["GFLOPS"] =
+      benchmark::Counter(uint64_t(state.iterations()) * 2 * M * N * K,
+                         benchmark::Counter::kIsRate);
+  }
+
+  int M;
+  int N;
+  int K;
+  at::Tensor A;
+  at::Tensor B;
+  at::Tensor C;
+};
+
+BENCHMARK_DEFINE_F(Gemm, Torch)(benchmark::State& state) {
+  for (auto _ : state) {
+    torch::mm_out(C, A, B);
```

Comment:
Minor: This is fine as it is. But this default of this benchmark completely fits in L2, which might not be representative. On the other hand, flushing cache might seem too big an interruption.

A possible middle-ground, is to have vectors of A, B and C, and 
```
mm_out(A[i % size_a], B[2 * i % size_b], C[3 * i % size_c]). 
```

Reply:
I definitely agree we want to have more problem sizes.  I started with 128x128x128 exactly because it fits in L2 :)